### PR TITLE
feat: added uuid validation rule

### PR DESCRIPTION
**Pull request by benswinburne**
## Proposed changes
Add the ability to validate UUIDs

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/adonisjs/validator/blob/master/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

## Further comments

Added an additional rule and corresponding tests. Shouldn't interfere with existing rules etc. Just stuck with the available options in `validator` so it won't validate UUID v1 or v2 or nil UUIDs.

### CODE REVIEW

File: src/Validations/string/uuid.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    ensureValidArgs(RULE_NAME, args)
+    const options = args[0]
+
+    return {
+      allowUndefineds: false,
+      async: false,
+      name: RULE_NAME,
+      compiledOptions: {
+        version: options && options.version,
```
Comment (thetutlage) [inline src/Validations/string/uuid.ts]: How about setting the version `4` here, instead of doing it at runtime?

**Review by thetutlage — CHANGES_REQUESTED** [review]

File: src/Validations/string/uuid.ts (hunk 1 above)
Comment (benswinburne) [reply src/Validations/string/uuid.ts]: Makes more sense 👍 Done.

**Comment by thetutlage** [comment]
Closed and re-opened to kick in circleci tests

**Comment by thetutlage** [comment]
Thanks for contributing. The PR was great 😄

**Pull request merged**